cores/arduino/stm32/analog.c

```diff
@@ -809,6 +1066,52 @@ void pwm_stop(PinName pin)
 }
 
 
+/**
+  * @brief  Conversion DMA transfer callback in non blocking mode 
+  * @param  hadc: ADC handle
+  * @retval None
+  */
+void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef *AdcHandle)
+{
+  /* Report to main program that ADC sequencer has reached its end */
```

Comment:
Because we're making sure that ADC_HandleTypeDef  is the first struct in the AdContext struct this is a trick to retrieve the AdcContext from the handle used in HAL.
you should be able to code as below:

AdcContext context = (AdcContext *) AdcHandle;
contex.cb(context.user_data);

Reply:
ok. Done. New release available.